### ISSUE TITLE
Right Sidebar says "No Row Selected"

### ISSUE THREAD

**Issue by jakob** [jakob/Postico#336]
(even though there clearly is a row selected)

I'm not sure how to reproduce this error. Might have something to do with the auto-disconnect setting.

**Comment by jakob** [comment]
I've managed to trigger this bug accidentally on macOS 10.10, but don't know how. I have not been able to reproduce it on macOS 10.12

**Comment by jakob** [comment]
I've been able to reproduce the problem again, however I'm not sure what is causing it. The problem results in `tableViewSelectionDidChange` no longer being called. [This Stack Overflow Answer](http://stackoverflow.com/a/19957975/322427) suggests that the problem could be that I'm calling `removeObserver:` incorrectly somewhere. That is probably the cause, but I haven't been able to find the exact call that's causing the problem yet.

**Comment by jakob** [comment]
I was able to track the error down. It can be triggered by shutting the server down, then pressing "Clear Filter" button twice, then starting the server again and reloading. Then the right sidebar becomes broken.

The problem was indeed an improper call of `removeObserver:`. I've fixed this specific call site by adding a nil-check. I've also audited the entire code base to make sure I don't have similar errors with `removeObserver:` in other places, and added some more nil checks to be safe.
